SU2_CFD/src/variables/CNEMOEulerVariable.cpp

```diff
@@ -343,4 +346,88 @@ bool CNEMOEulerVariable::Cons2PrimVar(su2double *U, su2double *V,
   return nonPhys;
 }
 
+void CNEMOEulerVariable::Prim2ConsVar(su2double *U, su2double *V) {
+
+  /*---Useful variables ---*/
+  vector<su2double> Energies, rhos;
+  rhos.resize(nSpecies,0.0);
+
+  /*--- Set densities and mass fraction ---*/
+  for (unsigned short iSpecies = 0; iSpecies < nSpecies; iSpecies++){
+    U[iSpecies]    = V[iSpecies];
+    rhos[iSpecies] = V[iSpecies];
+  }
+
+  /*--- Set momentum and compute v^2 ---*/
+  //TODO: geometry toolbox
+  su2double sqvel = 0.0;
+  for (unsigned short iDim = 0; iDim < nDim; iDim++){
+    U[nSpecies+iDim] = V[RHO_INDEX]*V[VEL_INDEX+iDim];
+    sqvel           += V[VEL_INDEX+iDim]*V[VEL_INDEX+iDim];
+  }
+
+  /*--- Set the fluidmodel and recompute energies ---*/
+  fluidmodel->SetTDStateRhosTTv( rhos, V[T_INDEX], V[TVE_INDEX]);
+  Energies = fluidmodel->ComputeMixtureEnergies();
+
+  /*--- Set conservative energies ---*/
+  U[nSpecies+nDim]   = V[RHO_INDEX]*(Energies[0]+0.5*sqvel);
+  U[nSpecies+nDim+1] = V[RHO_INDEX]*(Energies[1]);
+}
+
 void CNEMOEulerVariable::SetSolution_New() { Solution_New = Solution; }
+
+bool CNEMOEulerVariable::CheckNonPhys(su2double *V) {
+
+  su2double Tmin, Tmax, Tvemin, Tvemax;
+
+  /*--- Set booleans ---*/
+  bool nonPhys = false;
+
+  /*--- Set temperature clipping values ---*/
+  Tmin   = 50.0; Tmax   = 8E4;
+  Tvemin = 50.0; Tvemax = 8E4;
+
+  /*--- Check whether state makes sense ---*/
+  for (unsigned short iSpecies = 0; iSpecies < nSpecies; iSpecies++)
+    if (V[RHOS_INDEX+iSpecies] < 0.0) nonPhys = true;
+
+  if (V[P_INDEX] < 0.0) nonPhys = true;
+
+  if (V[T_INDEX] < Tmin || V[T_INDEX] > Tmax) nonPhys = true;
+
+  if (V[TVE_INDEX] < Tvemin || V[TVE_INDEX] > Tvemax) nonPhys = true;
+
+  if (V[A_INDEX] < 0.0 ) nonPhys = true;
+
+  return nonPhys;
+
+}
+
+su2double CNEMOEulerVariable::ComputeConsistentExtrapolation(su2double* V, su2double* dPdU, su2double* dTdU,
+                                                             su2double* dTvedU, su2double *val_eves,
+                                                             su2double *val_Cvves) {
+
+  //NOTE: TODO - this doesnt compute Cvves/ dPdU,etc.yet
+  su2double val_gamma;
+  vector<su2double> rhos, eves, cvves;
+
+  /*--- Rename density vector ---*/
+  rhos.resize(nSpecies,0.0);
+  for (unsigned short iSpecies=0; iSpecies < nSpecies; iSpecies++ ){
+    rhos[iSpecies] = V[iSpecies];
+  }
+
+  /*--- Set new fluid state ---*/
+  fluidmodel->SetTDStateRhosTTv(rhos, V[T_INDEX], V[TVE_INDEX]);

```

Comment:
The other reason to have this kind of function in the solver is that, when calling it in a loop you can declare the vectors outside of the loop which saves you the dynamic allocation / deallocation every time the function is called.

Reply:
just to clarify, why are we leaving dThing/dU and Cvves out for now ? don't we need it, for example, for Roe scheme ?

Reply:
In most cases, the computation is unneccessary, but most likely will be useful in the case of the implicit formulation.  I figured it would be easy to have these here now and remove in the next if necessary

Reply:
in most or in all of them? asking cause that will make the difference between this being totally functional or not

Reply:
so is it just for implicit ? or something else?